docs/03-extensions/adyen/how-to/front-commerce.mdx

```diff
@@ -0,0 +1,191 @@
+FRONT_COMMERCE_ADYEN_HMAC_KEY=the_hmac_key
+FRONT_COMMERCE_ADYEN_PREVIOUS_HMAC_KEY=the_previous_hmac_key
+FRONT_COMMERCE_ADYEN_STORE_PAYMENT_DETAILS=true
+```
+
+See our
+[extension reference](/docs/3.x/extensions/adyen/reference/environment-variables)
+for more information about these variables.
+
+## Register your Adyen payment components
+
+:::note
+
+This is a temporary way to setup payment components. We are aware that it is
+tedious. It will definitely change in the future, when Front-Commerce will
+support new extension points for Payments.
+
+:::
+
+1. Override the file that lets you register additional payments forms in
+   Front-Commerce
+
+   ```shell
+   mkdir -p app/theme/modules/Checkout/Payment/AdditionalPaymentInformation/
+   cp -u node_modules/@front-commerce/theme-chocolatine/theme/modules/Checkout/Payment/AdditionalPaymentInformation/getAdditionalDataComponent.js app/theme/modules/Checkout/Payment/AdditionalPaymentInformation/getAdditionalDataComponent.js
+   ```
+
+2. Register Adyen
+
+   ```diff title="app/theme/modules/Checkout/Payment/AdditionalPaymentInformation/getAdditionalDataComponent.js"
+   +import AdyenCheckout from "theme/modules/Checkout/Payment/AdditionalPaymentInformation/AdyenCheckout";
+
+   const ComponentMap = {};
+
+   const getAdditionalDataComponent = (method) => {
+   +  if (method.code.startsWith("adyen_")) {
+   +    return AdyenCheckout;
+   +  }
+     return ComponentMap[method.code];
+   };
+   ```
+
+3. Override the file that lets you register additional payments actions in
+   Front-Commerce
+
+   ```shell
+   mkdir -p app/theme/modules/Checkout/PlaceOrder
+   cp -u node_modules/@front-commerce/theme-chocolatine/theme/modules/Checkout/PlaceOrder/getAdditionalActionComponent.js app/theme/modules/Checkout/PlaceOrder/getAdditionalActionComponent.js
+   ```
+
+4. Register the Adyen action
+
+   ```diff title="app/theme/modules/Checkout/PlaceOrder/getAdditionalActionComponent.js"
+   import None from "theme/modules/Checkout/PlaceOrder/AdditionalAction/None";
+   +import Adyen from "theme/modules/Checkout/PlaceOrder/AdditionalAction/Adyen";
+
+   const ComponentMap = {};
+
+   const getAdditionalActionComponent = (paymentCode, paymentAdditionalData) => {
+   +  if (paymentCode.startsWith("adyen")) {
+   +    return Adyen;
+   +  }
+     return ComponentMap?.[paymentCode] ?? None;
+   };
+   ```
+
+## Register custom styles (optional)
+
+Developers can
+[customize Adyen drop-in UI styles using CSS](https://docs.adyen.com/checkout/drop-in-web/customization).
+
+You can add an optional stylesheet from Front-Commerce in your application to
+customize styles. The provided stylesheet reuses existing styles from the theme
+as much as possible for a good integration by default. You can override it if
+needed to adapt the UI as wanted.
+
+Add the following line to your your main `.scss` file to load these styles:
+
+```scss
+@import "theme/modules/Adyen/dropinCustomizations";
+```
```

Comment:
**thought:** should we do it in the stylesheet, or directly import the customization in the project from a component?

For instance it could be added in the `getAdditionalActionComponent.js` component

Reply:
Yes I was wondering the same. Having them already loaded could be practical for integrators, but might also result in unused styles if they don't want to override the classes.